libraries/wallet/wallet.cpp

```diff
@@ -2042,30 +2078,45 @@ fc::ecc::private_key wallet_api::derive_private_key(const std::string& prefix_st
             for( auto& key_weight_pair : account.owner.key_auths )
             {
                 for( auto& key : keys )
-                    FC_ASSERT( key_weight_pair.first != key, "Detected private owner key in memo field. Cancelling transaction." );
+                    GOLOS_CHECK_LOGIC(key_weight_pair.first != key, 
+                            logic_errors::detected_private_key_in_memo,
+                            "Detected ${type} private key in memo field",
+                            ("type","owner"));
```

Comment:
Why use constant string as substitution? Result will be always the same

Reply:
I mean why use
`"Detected ${type} private key in memo field", ("type","owner")`
vs simple
`"Detected owner private key in memo field"`

Reply:
> Passing this parameter this way not only performs substitution, but also adds this parameter with a value to the list of error properties

Ok, I see. If so, maybe put it in lambda or function to remove slightly reduce duplication and make this decision clearer.